### PR TITLE
clerk side handler

### PR DESCRIPTION
**Pull request by temaniarpit27**

### CODE REVIEW

**Review by ssandeep — APPROVED** [review]

**Comment by ssandeep** [comment]
Can you also `make lint` and fix them?

File: x/clerk/side_handler.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		case *types.MsgEventRecordRequest:
+			return PostHandleMsgEventRecord(ctx, k, *msg, sideTxResult)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type: %T", types.ModuleName, msg)
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+		}
+	}
+}
+
+func SideHandleMsgEventRecord(ctx sdk.Context, k keeper.Keeper, msg types.MsgEventRecordRequest, contractCaller helper.IContractCaller) (result abci.ResponseDeliverSideTx) {
```
Comment (anilCSE) [inline x/clerk/side_handler.go:52]: split this line

File: x/clerk/side_handler.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+	// get event log for topup
+	eventLog, err := contractCaller.DecodeStateSyncedEvent(chainParams.StateSenderAddress, receipt, msg.LogIndex)
+	if err != nil || eventLog == nil {
+		k.Logger(ctx).Error("Error fetching log from txhash")
+		return hmCommon.ErrorSideTx(hmCommon.CodeErrDecodeEvent)
+	}
+
+	if receipt.BlockNumber.Uint64() != msg.BlockNumber {
+		k.Logger(ctx).Error("BlockNumber in message doesn't match blocknumber in receipt", "MsgBlockNumber", msg.BlockNumber, "ReceiptBlockNumber", receipt.BlockNumber.Uint64())
```
Comment (anilCSE) [inline x/clerk/side_handler.go:78]: split this line

File: x/clerk/side_handler.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+			"MsgData", hmTypes.BytesToHexBytes(msg.Data),
+		)
+		return hmCommon.ErrorSideTx(hmCommon.CodeInvalidMsg)
+	}
+
+	result.Result = tmprototypes.SideTxResultType_YES
+	return
+}
+
+func PostHandleMsgEventRecord(ctx sdk.Context, k keeper.Keeper, msg types.MsgEventRecordRequest, sideTxResult tmprototypes.SideTxResultType) (*sdk.Result, error) {
```
Comment (anilCSE) [inline x/clerk/side_handler.go:110]: split this line

File: x/clerk/types/msg.go [hunk 4]
```diff
@@ -15,7 +14,7 @@ var _ sdk.Msg = &MsgEventRecordRequest{}
 // NewMsgEventRecord - construct state msg
 func NewMsgEventRecord(
 	from sdk.AccAddress,
-	txHash hmCommon.HeimdallHash,
+	txHash []byte,
```
Comment (anilCSE) [inline x/clerk/types/msg.go:17]: We shouldn't change this. Let's use `HeimdallHash`

**Review by anilCSE — CHANGES_REQUESTED** [review]

File: x/clerk/side_handler.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+// NewSideTxHandler returns a side handler for "topup" type messages.
+func NewSideTxHandler(k keeper.Keeper, contractCaller helper.IContractCaller) hmTypes.SideTxHandler {
+	return func(ctx sdk.Context, msg sdk.Msg) abci.ResponseDeliverSideTx {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case *types.MsgEventRecordRequest:
+			return SideHandleMsgEventRecord(ctx, k, *msg, contractCaller)
+		default:
+			return abci.ResponseDeliverSideTx{
+				Code: uint32(6), // TODO should be changed like `sdk.CodeUnknownRequest`
```
[the remark this answers is not in the document]
Comment (temaniarpit27) [reply x/clerk/side_handler.go:32]: we will address it later

File: x/clerk/types/msg.go [hunk 6]
```diff
@@ -48,7 +48,7 @@ func (msg MsgEventRecordRequest) ValidateBasic() error {
 		return sdkerrors.ErrUnknownRequest
 	}
 
-	if msg.TxHash.Empty() {
+	if len(msg.TxHash) == 0 {
```
Comment (anilCSE) [inline x/clerk/types/msg.go:51]: Can't we use .Empty here?

**Review by anilCSE — APPROVED** [review]
lgtm, just small nit

**Pull request merged**